src/components/DevconGrantsForm.js

```diff
@@ -0,0 +1,775 @@
+        : name === "eventLink"
+        ? isUrlValid()
+        : name === "city"
+        ? isCityValid()
+        : requiredFields.includes(name)
+        ? value !== ""
+        : true
+    setFormState(snapshot)
+  }
+
+  const handleSelectChange = selectedOption => {
+    const { name, value } = selectedOption
+    const snapshot = { ...formState }
+    snapshot[name].value = value
+    snapshot[name].isValid = value !== "" || !requiredFields.includes(name)
+    setFormState(snapshot)
+  }
+
+  const handleTouched = (event, field) => {
+    const name = field || event.target.name
+    const snapshot = { ...formState }
+    snapshot[name].isTouched = true
+    setFormState(snapshot)
+  }
+
+  const submitInquiry = async () => {
+    setFormState({ ...formState, isPending: true })
+    const stateFields = Object.keys(formState)
+    const formStateFields = stateFields.filter(field => formState[field].value)
+    const formData = {}
+    for (const field of formStateFields) {
+      formData[field] = formState[field].value
+    }
+
+    try {
+      const response = await fetch("/.netlify/functions/inquiry", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      })
+      setFormState({ ...formState, isPending: false })
+      if (response.status !== 200) {
+        addToast(formatMessage({ id: `page-devcon.error-submitting` }), {
+          appearance: "error",
+          autoDismiss: true,
+        })
+      } else {
+        addToast(formatMessage({ id: `page-devcon.success` }), {
+          appearance: "success",
+          autoDismiss: true,
+        })
+        navigate("/thanks/")
+      }
+    } catch (error) {
+      setFormState({ ...formState, isPending: false })
+      console.error(error)
+      addToast(formatMessage({ id: `page-devcon.error-submitting` }), {
+        appearance: "error",
+        autoDismiss: true,
+      })
+    }
+  }
+
+  const handleSubmit = event => {
+    event.preventDefault()
+    // Already be validated before this button is enabled
+    isFormValid() && submitInquiry()
+  }
+
+  const isFormValid = () => {
+    for (const field of requiredFields) {
+      if (!formState[field]?.isValid) return false
+    }
+    return formState.eventLink.isValid && formState.city.isValid
+  }
+
+  const isButtonDisabled = !isFormValid() || formState.isPending
+  const buttonText = formState.isPending
+    ? formatMessage({ id: `page-devcon.submitting` })
+    : formatMessage({ id: `page-devcon.submit` })
+
+  const RequiredError = () => <ErrorMessage>Field is required</ErrorMessage>
+
+  return (
+    <StyledForm onSubmit={handleSubmit}>
+      <StyledLabel>
+        <span>
+          <FormattedMessage id="page-devcon.contact-first-name" />{" "}
+          <Required>*</Required>
+        </span>
+        <Input
+          type="text"
+          name="firstName"
+          value={formState.firstName?.value}
```

Comment:
Curious, does this give you syntax highlighting issues in VSCode?

I suspected it was because only [TypeScript supports optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) but it looks like JS (on most browsers) does to: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
It's probably an extension causing this for me 🤔